### PR TITLE
[#17] EPA blow up polytop; [#16] SSID Contact Manifold

### PR DESCRIPTION
**Pull request by ilia-glushchenko**

### CODE REVIEW

File: Arion/include/Arion/SimpleShapeIntersection.hpp [hunk 1]
```diff
@@ -100,6 +99,33 @@ std::pair<glm::dvec3, glm::dvec3> CalculateContactPoints(SimpleShape const* a, S
 template <typename ShapeA, typename ShapeB>
 double CalculatePenetration(SimpleShape const* a, SimpleShape const* b, CacheBase* cache);
 
+/**
+ * @brief Calculates contact manifold for given shapes
+ *
+ * @attention Must be called strictly after the corresponding CalculateIntersection function call, otherwise result is undefined
```
Comment (MrDmitry) [inline Arion/include/Arion/SimpleShapeIntersection.hpp]: right now both `CalculateContactNormal` and `CalculateContactManifold` have to be called `strictly after the corresponding CalculateIntersection function call`
does `strictly` apply?

File: Arion/include/Arion/SimpleShapeIntersection.hpp [hunk 2]
```diff
@@ -863,11 +965,16 @@ inline glm::dvec3 CalculateContactNormal<Box, Plane>(SimpleShape const* a, Simpl
 
 /** Box, Plane CalculateContactPoints specialization */
 template <>
-inline std::pair<glm::dvec3, glm::dvec3>  CalculateContactPoints<Box, Plane>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
+inline ContactPoints  CalculateContactPoints<Box, Plane>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
```
Comment (MrDmitry) [inline Arion/include/Arion/SimpleShapeIntersection.hpp]: double space `ContactPoints  CalculateContactPoints`

File: Arion/include/Arion/SimpleShapeIntersectionDetector.hpp [hunk 3]
```diff
@@ -52,26 +52,42 @@ class SimpleShapeIntersectionDetector
...
-     * 
+     *
      * @return penetration depth
      */
     double CalculatePenetration(SimpleShape const* a, SimpleShape const* b);
 
+   /**
+    * @brief Calculates contact manifold for given shapes
+    *
+    * @attention Must be called strictly after the corresponding CalculateIntersection function call, otherwise result is undefined
```
Comment (MrDmitry) [inline Arion/include/Arion/SimpleShapeIntersectionDetector.hpp]: same question as before - strictness conflict with `CalculateContactNormal`

**Review by MrDmitry — CHANGES_REQUESTED** [review]

**Review by MrDmitry — APPROVED** [review]

File: Arion/include/Arion/SimpleShapeIntersection.hpp [hunk 4]
```diff
@@ -573,7 +655,7 @@ template <>
 inline bool CalculateIntersection<Plane, Box>(SimpleShape const* a, SimpleShape const* b, CacheBase* cacheBase)
 {
     auto plane = static_cast<Plane const*>(a);
-    auto box = static_cast<Box const*>(b);
+    auto const box = static_cast<Box const*>(b);
```
Comment (MrDmitry) [inline Arion/include/Arion/SimpleShapeIntersection.hpp]: should the `plane` be `const` as well?

Comment (ilia-glushchenko) [reply Arion/include/Arion/SimpleShapeIntersection.hpp]: it should

**Review by MrDmitry — APPROVED** [review]

**Comment by t0mgit** [comment]
nice.

**Comment by t0mgit** [comment]
approve

[Approved with PullApprove](https://pullapprove.com/Godlike/Arion/pull-request/18/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Godlike/Arion)

**Pull request merged**